src/main/java/org/apache/commons/lang3/CharUtils.java

```diff
@@ -114,6 +114,33 @@ public static Character toCharacterObject(final String str) {
         return StringUtils.isEmpty(str) ? null : Character.valueOf(str.charAt(0));
     }
 
+    /**
+     * <p>Converts the String to a Character using the first character, returning a
+     *  default value if {@code null} or empty is passed in.</p>
+     *
+     * <p>For ASCII 7 bit characters, this uses a cache that will return the
+     * same Character object each time.</p>
+     *
+     * <pre>
+     *   CharUtils.toCharacterObject(null, 'A') = 'A'
+     *   CharUtils.toCharacterObject("", 'A')   = 'A'
+     *   CharUtils.toCharacterObject("A", 'D')  = 'A'
+     *   CharUtils.toCharacterObject("BA", 'D') = 'B'
+     * </pre>
+     *
+     * @param str  the character to convert
+     * @param defaultValue the value to use if the str is null or empty.
+     * @return the Character value of the first letter of the String or the default if null.
+     * @since 3.13.0
+     */
+    public static Character toCharacterObject(final String str, final Character defaultValue) {
+      final Character character = toCharacterObject(str);
+      if (character == null){
+          return defaultValue;
+      }
+      return character;
+    }
```

Comment:
I think the rest of the code is using 4 spaces (you used 4 spaces in the test).

Reply:
Better like this IMO: `return StringUtils.isEmpty(str) ? defaultValue : Character.valueOf(str.charAt(0));`

Reply:
> I think if you still uses ternary conditions in 2023, you are a really bad developer. For the love of god, understand ternaries are unreadable and are a very bad practice to show junior developers.

The ternary expression is an uncontroversial feature of the Java language as well as of most programming languages. This is the first time I've ever read anyone express dismay at its usage. For a great overview of best practices, I recommend Josh Block's Effective Java.